libraries/cyclestreets-view/src/main/java/net/cyclestreets/LiveRideActivity.kt

```diff
@@ -19,15 +26,39 @@ import net.cyclestreets.util.hasPermission
 
 private val TAG = Logging.getTag(LiveRideActivity::class.java)
 
-class LiveRideActivity : Activity() {
+class LiveRideActivity : Activity(), ServiceConnection, LiveRideOverlay.Locator {
     private lateinit var map: CycleMapView
+    private lateinit var liveride: LiveRideService.Binding
 
     override fun attachBaseContext(newBase: Context) {
         // Allows the use of Material icon library, see https://github.com/mikepenz/Android-Iconics
         super.attachBaseContext(IconicsContextWrapper.wrap(newBase))
     }
 
-    // No need to override onCreate - the map is initialized in onResume
+    override fun onServiceConnected(className: ComponentName, binder: IBinder) {
+        liveride = binder as LiveRideService.Binding
+
+        if (!liveride.areRiding())
+            liveride.startRiding()
+    }
+
+    override fun onServiceDisconnected(className: ComponentName) {}
+
+    override fun lastLocation(): Location? {
+        return liveride.lastLocation()
+    }
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val intent = Intent(this, LiveRideService::class.java)
+        this.bindService(intent, this, Context.BIND_AUTO_CREATE)
+    }
+
+    public override fun onDestroy() {
+        this.unbindService(this)
```

Comment:
Do we need to call
```kotlin
liveride?.stopRiding
```
here?  Previously we did (see https://github.com/cyclestreets/android/pull/321/files#diff-78f316180ff253b501bc22d4a1cd8799L76)

Reply:
Yes.  The service will stop itself when it's destroyed, which is triggered by unbinding, but it's probably better to shut it down first.